src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,526 @@
+
+import numpy as np
+from .gridsample import _gridsample
+from .freesample import _freesample
+
+class LintSampler:
+    """Draw sample(s) from density function defined for a list of cells (that may or may not be in a single grid)
+
+    Examples
+    --------
+    
+    These examples demonstrate the multiple ways to use ``LintSampler``. In
+    each case, we'll just generate densities from a uniform distribution, but
+    in general they might come from any arbitrary density function.
+
+    1. A single sample from a 1D grid. The grid spans x=0 to x=10, and has 32
+    cells (so 33 edges). 
+    
+    >>> cells = np.linspace(0, 10, 33)
+    >>> def pdfrandom(X): return np.random.uniform(size=X.shape[0])
+    >>> LintSampler(pdfrandom,cells).sample()
+    6.984134639227398
+
+    This returns a single scalar: the sampling point within the grid.
+    
+    2. Multiple samples from a 1D grid (same grid as previous example).
+    
+    >>> cells = np.linspace(0, 10, 33)
+    >>> def pdfrandom(X): return np.random.uniform(size=X.shape[0])
+    >>> LintSampler(pdfrandom,cells).sample(4)
+    array([8.16447008, 5.30536088, 8.96135879, 7.73572977])
+
+    This returns a 1D array: the ``N_samples`` sampling points within the grid.
+
+    3. Single sample from a k-D grid. In this case we'll take a 2D grid, with
+    32 x 64 cells (so 33 gridlines along one axis and 65 along the other, and
+    33x65=2145 intersections with known densities).
+    
+    >>> x = np.linspace(0, 10, 33)
+    >>> y = np.linspace(100, 200, 65)
+    >>> cells = (x,y)
+    >>> def pdfrandom(X): return np.random.uniform(size=X.shape[0])
+    >>> LintSampler(pdfrandom,cells).sample()
+    array([  7.67294632, 190.45302915])
+
+    This returns a 1D array: the single k-D sampling point within the grid.
+
+    4. Multiple samples from a k-D grid (same grid as previous example).
+    
+    >>> x = np.linspace(0, 10, 33)
+    >>> y = np.linspace(100, 200, 65)
+    >>> cells = (x,y)
+    >>> def pdfrandom(X): return np.random.uniform(size=X.shape[0])
+    >>> LintSampler(pdfrandom,cells).sample(5)
+    array([[1.35963966e-01, 1.38182930e+02],
+           [6.52704300e+00, 1.63109912e+02],
+           [4.35226761e+00, 1.49753235e+02],
+           [3.56093155e+00, 1.48548481e+02],
+           [1.31163401e+00, 1.59335676e+02]])
+
+    This returns a 2D array, shape ``N_samples`` x k: the ``N_samples`` k-D
+    samples within the grid.
+    """
+
+    def __init__(self, pdf, cells=(), rnginput=None, vectorizedpdf=False):
```

Comment:
Here a default value (empty tuple) is given to cells, but (I believe) this default value would throw an error, should we remove the default value?
```suggestion
    def __init__(self, pdf, cells, rnginput=None, vectorizedpdf=False):
```